docs/src/development/variables.md

```diff
@@ -73,25 +75,54 @@ If you omit the variable `--value` from the command line as above, you will be p
 
+#### Variables available only at build time
+
 The following variables are only available at build time, and may be used in a build hook:
 
 * **PLATFORM_OUTPUT_DIR**: The output directory for compiled languages at build time. Will be equivalent to `PLATFORM_APP_DIR` in most cases.
 
+#### Variables available during builds and at runtime
+
 The following variables are available at both runtime and at build time, and may be used in a build hook:
 
-* **PLATFORM_APP_DIR**: The absolute path to the application directory.
-* **PLATFORM_APPLICATION**: A base64-encoded JSON object that describes the application. It maps the content of the `.platform.app.yaml` that you have in Git and it has a few subkeys.
-* **PLATFORM_APPLICATION_NAME**: The name of the application, as configured in the `.platform.app.yaml` file.
 * **PLATFORM_PROJECT**: The ID of the project.
 * **PLATFORM_TREE_ID**: The ID of the tree the application was built from. It's essentially the SHA hash of the tree in Git.  If you need a unique ID for each build for whatever reason this is the value you should use.
 * **PLATFORM_VARIABLES**: A base64-encoded JSON object which keys are variables names and values are variable values (see below).  Note that the values available in this structure may vary between build and runtime depending on the variable type as described above.
 * **PLATFORM_PROJECT_ENTROPY**: A random, 56-character value created when the project is first created, which is then stable throughout the project's life. This can be used for Drupal hash salt, Symfony secret, or other similar values in other frameworks.
+* **PLATFORM_APP_DIR**: The absolute path to the application directory.
+* **PLATFORM_APPLICATION_NAME**: The name of the application, as configured in the `.platform.app.yaml` file.
+* **PLATFORM_APPLICATION**: A base64-encoded JSON object that describes the application. It maps certain attributes from the `.platform.app.yaml` that you have in Git and it has a few subkeys.
+
+`PLATFORM_APPLICATION` is a special case to keep in mind in how it differs between the build and runtime container. Every environment's build is associated with a configuration ID that uniquely identifies it. It's this ID which allows us to reuse builds on merges. The ID itself is a product of your application code, as well as its configuration for Platform.sh in `.platform.app.yaml`. 
+
+Not every attribute in `.platform.app.yaml` is relevant to builds - that is, not every attribute is accessible to the build container in `PLATFORM_APPLICATION`, nor will commits which edit those attributes result in a full rebuild of the application when pushed. Because of this, not all attributes defined in your `.platform.app.yaml` file will be accessible at build time from `PLATFORM_APPLICATION`, only those actually relevant to builds. Some of those attributes that will **not** be available in `PLATFORM_APPLICATION` during buids include:
```

Comment:
```suggestion
Not every attribute in `.platform.app.yaml` is relevant to builds - that is, not every attribute is accessible to the build container in `PLATFORM_APPLICATION`, nor will commits which edit those attributes result in a full rebuild of the application when pushed. Because of this, not all attributes defined in your `.platform.app.yaml` file will be accessible at build time from `PLATFORM_APPLICATION`, only those actually relevant to builds. Some of those attributes that will **not** be available in `PLATFORM_APPLICATION` during builds include:
```

Reply:
@maria-webb this was redundant, but I ended up rewriting/keeping the second mention instead.